### ISSUE TITLE
GSoD ideas for the book 

### ISSUE THREAD

**Issue by omar-3** [CircuitVerse/Interactive-Book#270]
Hi I already introduced myself on slack, so here is my two cents about what I'm supposed to do on GSoC, **need follow up from you guys** :)

About the project, point one and point two of the deliverables are clear. About point three and the addition of new content, I was thinking about adding sections about the following:

1. Microarchitecture which includes an explanation for Single-Cycle Processors and Multi-Cycle Processors and Pipelined Processors, and after that memory and caches and virtual Memory.

2. After the above point, we could start writing about assembly and specific architectures? I'm thinking about ARM as I worked with it for a considerable time, but I would prefer x86 architecture as it is more mainstream.

3. After all of that, we could have 3-4 sections about FPGA/ASIC design and basic principals.

About the last point of improvement of circuit interactions, I don't fully understand what you mean? but if you mean making the sketches more javascript-y and with labels and animations, it would be okay with me.

--- 

What should I do next? I don't have any writing samples, so I would go through the book in details and try to edit stuff and refine things. Could I begin on the new sections right away? What you guys think?

**Comment by eKuG** [comment]
Are you comfortable writing an simulating of ARM x86? Can you write some organic content on FPGAs?

**Comment by eKuG** [comment]
If you want any help, let me know

**Comment by omar-3** [comment]
> Are you comfortable writing an simulating of ARM x86? Can you write some organic content on FPGAs?

yes I guess we can write about the structure and simulation of ARM x86, you mean that write?, and I guess having original content on FPGAs is absolutely doable, we should begin from the very beginning right? 

> If you want any help, let me know

For sure, thank you

**Comment by eKuG** [comment]
@omar-3 If you need any help on this, please do let me know. I have a fairly good grasp on Digital Design. I guess, I can help you in the most parts

**Comment by Me-Priyank** [comment]
Outdated.
